### DISCUSSION TITLE
Great Lakes Marine Forecast

### DISCUSSION THREAD

**Discussion by elkusbry** [weather-gov/api#384] [Q&A]
I'm looking for the api to access the wind and wave forecast on the Great Lakes.  Is there access via json (key/value pairs)? (i.e. not the text block delivered from https://api.weather.gov/products/types/GLF which requires parsing) 🤞

Text block example: "productText" in https://api.weather.gov/products/7081afa5-0774-4f24-8bc0-0b43386cd7fb

**Comment by StephenClouse** [comment]
We do not currently have support for marine forecasts in API.

**Comment by elkusbry** [comment]
Is there another way to get to the raw data used to generate the data in the productText text block for example: https://api.weather.gov/products/7081afa5-0774-4f24-8bc0-0b43386cd7fb which don't require parsing?

I'm not sure what you mean by support your comment above?  It looks like NSH and GLF are part of the API products?  Can you explain what you mean please?

**Comment by StephenClouse** [comment]
What I mean is, we do not currently offer a data-driven forecast for marine areas. This is only functional for land points; e.g., https://api.weather.gov/gridpoints/LOT/74,72/forecast  The API will show a forecast for marine points but it will not contain marine-specific elements; it treats all points as land. Properly supporting marine in API is a future goal.

Some raw marine data is accessible via the gridpoint data endpoint; e.g. https://api.weather.gov/gridpoints/OKX/38,25  However, interpreting this data is not something we can currently offer support for. There is also currently a bug preventing Great Lakes marine points from resolving to their correct forecast office; this will be fixed in a future release.

We do have all the text products in API, as you noted, but we only offer them in their original text format. We do not parse them in any way.
